pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -122,17 +124,21 @@ func (p *LaunchTemplateProvider) Get(ctx context.Context, constraints *v1alpha1.
 }
 
 func (p *LaunchTemplateProvider) ensureLaunchTemplate(ctx context.Context, options *launchTemplateOptions) (*ec2.LaunchTemplate, error) {
+	// Ensure that multiple threads don't attempt to create the same launch template
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
```

Comment:
nit: you can embed sync.mutex directly to the struct and can do this-
```
p.Lock()
defer p.Unlock()
```
not a blocked